### PR TITLE
Concrete::getById() must return only Concrete objects

### LINKED ISSUE #13517: Concrete::getById() must return only Concrete objects
### Improvement description

v11 follow-up to #13508

### PR DESCRIPTION
**Pull request by blankse**
Fix https://github.com/pimcore/pimcore/issues/13517

### CODE REVIEW

File: doc/Development_Documentation/23_Installation_and_Upgrade/09_Upgrade_Notes/README.md [hunk 1]
```diff
@@ -83,6 +83,7 @@ Please make sure to set your preferred storage location ***before*** migration.
 - [Config] The config node `pimcore.admin` and related parameters are moved to AdminBundle directly under `pimcore_admin` node. Please adapt your parameter usage accordingly eg. instead of `pimcore.admin.unauthenticated_routes`, it should be `pimcore_admin.unauthenticated_routes`
 - [DataObjects] Removed deprecated preview url in class editor.
 - [DataObjects] Removed sql filter functionality for data object grid
+- [DataObjects] Loading non-Concrete objects with the Concrete class is not any longer possible
```
Comment (jdreesen) [inline doc/Development_Documentation/23_Installation_and_Upgrade/09_Upgrade_Notes/README.md]:
```suggestion
- [DataObjects] Loading non-Concrete objects with the Concrete class is no longer possible
```

File: models/DataObject/AbstractObject.php [hunk 2]
```diff
@@ -490,23 +490,9 @@ public static function getList($config = [])
 
     /**
      * @internal
-     *
-     * @param AbstractObject $object
-     *
-     * @return bool
      */
-    protected static function typeMatch(AbstractObject $object)
+    protected static function typeMatch(AbstractObject $object): bool
```
Comment (dvesh3) [inline models/DataObject/AbstractObject.php:494]: Can we also add a simple tests in `ObjectTest`?

```php
    /**
     * Verifies that loading only Concrete object from Concrete::getById().
     */
    public function testConcreteLoading()
    {
        $concreteObject = TestHelper::createEmptyObject(type: DataObject\Concrete::class);
        $loadedConcrete = DataObject\Concrete::getById($concreteObject->getId(), ['force' => true]);

        $this->assertIsObject($loadedConcrete, 'Loaded Concrete should be an object.');

        $nonConcreteObject = TestHelper::createObjectFolder();
        $loadedNonConcrete = DataObject\Concrete::getById($nonConcreteObject->getId(), ['force' => true]);
        
        $this->assertNull($loadedNonConcrete, 'Loaded Concrete should be null.');
    }
```

Comment (blankse) [reply models/DataObject/AbstractObject.php:494]: @dvesh3 I added the test https://github.com/pimcore/pimcore/pull/13548/commits/5a2bf5788e50f7417400a8860b48ba46f9de61f3

**Comment by dvesh3** [comment]
@blankse thanks a ton!

**Pull request merged**